charts/consul/templates/server-config-configmap.yaml

```diff
@@ -32,6 +32,11 @@ data:
       },
       "recursors": {{ .Values.global.recursors | toJson }},
       "retry_join": ["{{template "consul.fullname" . }}-server.{{ .Release.Namespace }}.svc:{{ .Values.server.ports.serflan.port }}"],
+      {{- if .Values.global.peering.enabled }}
+      "peering": {
+        "enabled": true
+      },
+      {{- end }}
```

Comment:
we only want to provide `peering` configuration if the helm value is true and omit it otherwise. This is so that we're compatible with older consul versions.